tests/BenchmarkDotNet.Analyzers.Tests/AnalyzerTests/Attributes/ArgumentsAttributeAnalyzerTests.cs

```diff
@@ -938,6 +1092,63 @@ static IEnumerable<string> GenerateData()
                 ( "typeof(string)", "System.Type" ),
                 ( "DummyEnum.Value1", "DummyEnum" )
             ];
+
+            public static IEnumerable<ValueTupleDouble<string, string>> ConstantValuesAndTypes =>
+            [
+                ( "true", "bool" ),
+                ( "(byte)123", "byte" ),

```

Comment:
If this is testing the case of `[Params(123)] public byte x;`, I think it shouldn't have the cast (same for others). Or maybe another case to test both with and without the cast.

Reply:
No, these were to test precise values of constants. The cast on the `byte` is because C# lacks a suffix for `byte` literals for some reason. The implicit integer to byte conversion is tested in `Providing_integer_value_types_within_target_type_range_should_not_trigger_diagnostic`.